deployment/ccip/deploy.go

```diff
@@ -485,226 +585,214 @@ func DeployChainContracts(
+					chain.Client,
+					rmnRemoteContract.Address(),
+				)
+				return ContractDeploy[*rmn_proxy_contract.RMNProxyContract]{
+					rmnProxyAddr, rmnProxy, tx, deployment.NewTypeAndVersion(ARMProxy, deployment.Version1_0_0), err2,
+				}
+			})
+		if err != nil {
+			e.Logger.Errorw("Failed to deploy RMNProxy", "err", err)
+			return err
+		}
+		e.Logger.Infow("deployed new RMNProxy", "addr", rmnProxy.Address)
+		rmnProxyContract = rmnProxy.Contract
+	} else {
+		e.Logger.Infow("rmn proxy already deployed", "addr", chainState.RMNProxy.Address)
+	}
+	if chainState.TestRouter == nil {
+		testRouterContract, err := deployContract(e.Logger, chain, ab,
+			func(chain deployment.Chain) ContractDeploy[*router.Router] {
+				routerAddr, tx2, routerC, err2 := router.DeployRouter(
+					chain.DeployerKey,
+					chain.Client,
+					weth9Contract.Address(),
+					rmnProxyContract.Address(),
+				)
+				return ContractDeploy[*router.Router]{
+					routerAddr, routerC, tx2, deployment.NewTypeAndVersion(TestRouter, deployment.Version1_2_0), err2,
+				}
+			})
+		if err != nil {
+			e.Logger.Errorw("Failed to deploy test router", "err", err)
+			return err
+		}
+		e.Logger.Infow("deployed test router", "addr", testRouterContract.Address)
+	} else {
+		e.Logger.Infow("test router already deployed", "addr", chainState.TestRouter.Address)
+	}
+
+	// if router is not already deployed, we deploy it
+	if chainState.Router == nil {
+		routerContract, err := deployContract(e.Logger, chain, ab,
+			func(chain deployment.Chain) ContractDeploy[*router.Router] {
+				routerAddr, tx2, routerC, err2 := router.DeployRouter(
+					chain.DeployerKey,
+					chain.Client,
+					weth9Contract.Address(),
+					rmnProxyContract.Address(),
+				)
+				return ContractDeploy[*router.Router]{
+					routerAddr, routerC, tx2, deployment.NewTypeAndVersion(Router, deployment.Version1_2_0), err2,
+				}
+			})
+		if err != nil {
+			e.Logger.Errorw("Failed to deploy router", "err", err)
+			return err
+		}
+		e.Logger.Infow("deployed router", "addr", routerContract.Address)
+	} else {
+		e.Logger.Infow("router already deployed", "addr", chainState.Router.Address)
+	}
+	nmContract := chainState.NonceManager
+	if chainState.NonceManager == nil {
+		nonceManager, err := deployContract(e.Logger, chain, ab,
+			func(chain deployment.Chain) ContractDeploy[*nonce_manager.NonceManager] {
+				nonceManagerAddr, tx2, nonceManager, err2 := nonce_manager.DeployNonceManager(
+					chain.DeployerKey,
+					chain.Client,
+					[]common.Address{}, // Need to add onRamp after
+				)
+				return ContractDeploy[*nonce_manager.NonceManager]{
+					nonceManagerAddr, nonceManager, tx2, deployment.NewTypeAndVersion(NonceManager, deployment.Version1_6_0_dev), err2,
+				}
+			})
+		if err != nil {
+			e.Logger.Errorw("Failed to deploy nonce manager", "err", err)
+			return err
+		}
+		e.Logger.Infow("Deployed nonce manager", "addr", nonceManager.Address)
+		nmContract = nonceManager.Contract
```

Comment:
instead of creating a new variable, let's just update `chainState.NonceManager` to maintain `chainState` as single source of truth

Same with the other contracts

Reply:
why not new var? seems clean. Also if you are deploying new noncemanager chain state won't be updated right away. only the addressbook is updated after deployment

Reply:
For someone updating this file in the future I can see friction for them trying to know which contracts to use `chainState.<Contract>` or the contract variable. 

Alternatively, initialize all these contract variables at the top of the file and initialize them to chainstate, and only use those new variables in the function